compiler/optimizer/pruner.go

```diff
@@ -145,10 +149,61 @@ func reverseComparator(op string) string {
 }
 
 func newMetadataPruner(pred dag.Expr) dag.Expr {
-	e, ok := pred.(*dag.BinaryExpr)
-	if !ok {
+	switch e := pred.(type) {
+	case *dag.BinaryExpr:
+		return metaPrunerBinaryExpr(e)
+	case *dag.RegexpSearch:
+		this, ok := e.Expr.(*dag.This)
+		if !ok {
+			return nil
+		}
+		prefix := regexpPrefix(e.Pattern)
+		maxPrefix := regexpMaxPrefix(prefix)
+		if prefix == "" || maxPrefix == "" {
+			return nil
+		}
+		min := &dag.Literal{Kind: "Literal", Value: zson.QuotedString(prefix)}
+		max := &dag.Literal{Kind: "Literal", Value: zson.QuotedString(maxPrefix)}
+		return dag.NewBinaryExpr("and",
+			compare("<", min, &dag.This{Kind: "This", Path: append(this.Path, "min")}),
+			compare(">=", max, &dag.This{Kind: "This", Path: append(this.Path, "max")}))
```

Comment:
If len(this.Path) < cap(this.Path), then both calls to append here will write to the same underlying array with the second call overwriting "min" with "max".
```suggestion
			compare("<", min, &dag.This{Kind: "This", Path: append(slices.Clone(this.Path), "min")}),
			compare(">=", max, &dag.This{Kind: "This", Path: append(slices.Clone(this.Path), "max")}))
```